src/borda_clj/client.clj

```diff
@@ -16,30 +17,52 @@
         (do (print "borda buffer full, discarding measurement" dimensions values)
             measurements))))) ; buffer full
 
+(defn reduce-reports [in-ch out-ch kill-ch max-buffer-size]
+  (go-loop [measurements {}]
+    (let [[v ch] (alts! [in-ch [out-ch measurements]])]
+      (cond
+        (= ch out-ch) (recur {})
+        (nil? v) (close! kill-ch)
+        :else (recur (apply collect measurements max-buffer-size v))))))
+
+(defn flush [measurements-ch send on-error]
+  (let [measurements (<!! measurements-ch)]
+    (when (not-empty measurements)
+      (try
+        (send measurements)
+        (catch Throwable e
+          (on-error measurements e))))))
+
+(defn flush-measurements [report-ch measurements-ch kill-ch interval send on-send-error]
```

Comment:
Since I extracted `flush` into the top-level it would perhaps be better to call this `periodically-flush-measurements` or `go-flush-measurements` or something.

Reply:
I went with the `go-*` convention and used the same for `reduce-reports`.